### PR TITLE
Add return statement to `HederaTokenService.redirectForToken()`

### LINKED ISSUE #298: Add return statement to `HederaTokenService.redirectForToken()`
### Problem

Currently, there is no return statement in `HederaTokenService.redirectForToken()`, so we can not call that contract from an external contract and get the result.

### Solution

Add a return statement with the result.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
**Description**:

Currently, there is no return statement in HederaTokenService.redirectForToken(), so we can not call that contract from an external contract and get the result.

**Related issue(s)**:

Fixes #298 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Nice catch.
1 question

File: contracts/hts-precompile/IHederaTokenService.sol [hunk 1]
```diff
@@ -797,5 +797,5 @@ interface IHederaTokenService {
 
     /// Initiates a Redirect For Token
     /// @param encodedFunctionSelector The function selector from the ERC20 interface + the bytes input for the function called
-    function redirectForToken(address token, bytes memory encodedFunctionSelector) external;
+    function redirectForToken(address token, bytes memory encodedFunctionSelector) external returns (int64 responseCode, bytes memory response);
```
[the remark this answers is not in the document]
Comment (natanasow) [reply contracts/hts-precompile/IHederaTokenService.sol:800]: It was just incorrect. Nothing to change on services :sunglasses: .

Comment (Nana-EC) [inline contracts/hts-precompile/IHederaTokenService.sol:800]: nit: update function comments with return details

File: contracts/hts-precompile/HederaTokenService.sol [hunk 2]
```diff
@@ -660,11 +660,12 @@ abstract contract HederaTokenService {
     /// Redirect for token
     /// @param token The token address
     /// @param encodedFunctionSelector The function selector from the ERC20 interface + the bytes input for the function called
-    function redirectForToken(address token, bytes memory encodedFunctionSelector) external {
-        (bool success, bytes memory data) = precompileAddress.call(
+    function redirectForToken(address token, bytes memory encodedFunctionSelector) external returns (int responseCode, bytes memory response) {
```
Comment (Nana-EC) [inline contracts/hts-precompile/HederaTokenService.sol:663]: nit: update function comments with return

**Review by Nana-EC — APPROVED** [review]
LG.
Unblocking but also noted nit comments

**Pull request merged**